sh/sync.sh

```diff
@@ -0,0 +1,670 @@
+# Remove a sync job information from database
+# Globals: 2 variables
+#   SYNC_LIST, SYNC_DETAIL_DIR
+# Arguments: 1
+#   ${1} = pid number of a sync job
+# Result: read description
+###################################################
+_remove_job() {
+    unset input_remove_job local_folder_remove_job drive_folder_remove_job new_list_remove_job
+    pid_remove_job="${1}" && input_remove_job="$(grep "${pid_remove_job}" "${SYNC_LIST}" || :)"
+
+    _tmp="${input_remove_job#*"|:_//_:|"}" && local_folder_remove_job="${_tmp%%"|:_//_:|"*}"
+    drive_folder_remove_job="${input_remove_job##*"|:_//_:|"}"
+    new_list_remove_job="$(grep -v "${pid_remove_job}" "${SYNC_LIST}" || :)"
+    printf "%s\n" "${new_list_remove_job}" >| "${SYNC_LIST}"
+
+    rm -rf "${SYNC_DETAIL_DIR:?}/${drive_folder_remove_job}${local_folder_remove_job}"
+    # Cleanup dir if empty
+    [ -z "$(find "${SYNC_DETAIL_DIR:?}/${drive_folder_remove_job}" -type f)" ] && rm -rf "${SYNC_DETAIL_DIR:?}/${drive_folder_remove_job}"
+    return 0
+}
+
+###################################################
+# Kill a sync job and do _remove_job
+# Globals: 1 function
+#   _remove_job
+# Arguments: 1
+#   ${1}" = pid number of a sync job
+# Result: read description
+###################################################
+_kill_job() {
+    pid_kill_job="${1}"
+    kill -9 "${pid_kill_job}" 2> /dev/null 1>&2 || :
+    _remove_job "${pid_kill_job}"
+    printf "Killed.\n"
+}
+
+###################################################
+# Show total no of sync jobs running
+# Globals: 1 variable, 2 functions
+#   Variable - SYNC_LIST
+#   Functions - _get_job_info, _remove_job
+# Arguments: 1
+#   ${1}" = v/verbose: Prints extra information ( optional )
+# Result: read description
+###################################################
+_show_jobs() {
+    unset list_show_job pid_show_job no_task_show_job
+    total_show_job=0 list_show_job="$(grep -v '^$' "${SYNC_LIST}" || :)"
+    printf "%s\n" "${list_show_job}" >| "${SYNC_LIST}"
+
+    while read -r line <&4; do
+        if [ -n "${line}" ]; then
+            _tmp="${line%%"|:_//_:|"*}" && pid_show_job="${_tmp##*: }"
+            _get_job_info "${pid_show_job}" "${1}" "${line}"
+            { [ "${RETURN_STATUS}" = 1 ] && _remove_job "${pid_show_job}"; } || { $((total_show_job += 1)) && no_task_show_job="printf"; }
+        fi
+    done 4< "${SYNC_LIST}"
+
+    printf "\nTotal Jobs Running: %s\n" "${total_show_job}"
+    [ -n "${1}" ] && "${no_task_show_job:-:}" "For more info: %s -j/--jobs v/verbose\n" "${0##*/}"
+    return 0
+}
+
+###################################################
+# Setup required variables for a sync job
+# Globals: 1 Variable
+#   SYNC_DETAIL_DIR
+# Arguments: 1
+#   ${1}" = Local folder name which will be synced
+# Result: read description
+###################################################
+_setup_loop_variables() {
+    folder_setup_loop_variables="${1}" drive_folder_setup_loop_variables="${2}"
```

Comment:
Codacy found an issue: [Remove '$' or use '_=$((expr))' to avoid executing output.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5865418)